### DISCUSSION TITLE
Per-device volume limiter

### DISCUSSION THREAD

**Discussion by ma-er233** [File-New-Project/EarTrumpet#1509] [Ideas]
Having a feature to limit the volume on each device would be nice.

For example, any volume level above 30% on my earbuds is instant deafening. So the latter half of the volume range is essentially useless for this device. Is it possible to both limit the maximum volume and remap the current 30% as the new 100% in order to have a more granular control? Another example would be my laptop built-in speakers. They only start to produce sound at around 10%, and anything above 85% is heavily distorted. It would be much easier to use if I can make the current 10-85% range show up as 0-100%.

The logarithmic volume scale option largely solved the issue in my first example. Though I would still appreciate an option to limit the maximum volume, in case I accidentally crank it to high and detonate my eardrums.

Thanks!

**Comment by dolodobendan** [comment]
I searched for preamp and got no results. But yes, please, this.

https://github.com/File-New-Project/EarTrumpet/discussions/1529#discussion-6431662
